### ISSUE TITLE
No need for offstrategy compaction for tablet repair

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylladb#17384]
Since each tablet is repaired in a single token range, we don't need to collect repair's output sstable(s) in the tablet's compaction_group maintenance sstable set and run off-strategy compaction.

In stead, we can add the repair output sstable(s) directly into the main sstable set and rely on regular compaction.
In particular is meaningful for TWCS, where with offstrategy compaction disabled on this path, repair will segregate it's output into at most one sstable per time-window while with offstrategy compaction enabled, it will create a single sstable for the token-range and that sstable requires offstrategy compaction to be segregated into the time-windows.

**Comment by denesb** [comment]
@bhalevy this is an optimization that we can push out to 6.1, no?

**Comment by bhalevy** [comment]
> @bhalevy this is an optimization that we can push out to 6.1, no?

Yes.  In hindsight, we should have started with this from day one, but now it missed the train.

**Comment by bhalevy** [comment]
Given https://github.com/scylladb/scylla-dtest/pull/4161, we should reconsider doing this for 6.0.
It would be simpler than making sure offstrategy compaction is triggered for tablets,
assuming a hard coded, unconditional implementation that chooses offstrategy::no in both repair master and followers when the table supports tablets.

**Comment by mykaul** [comment]
@denesb - can you add priority here? It's unclear if it's nice to have, should have or must have - for 6.1 or defer to later.

**Comment by denesb** [comment]
It is an optimization, so a nice-to-have.

**Comment by dani-tweig** [comment]
Is this issue still relevant?
Can we change the milestone to Backlog?

**Comment by raphaelsc** [comment]
> Is this issue still relevant? Can we change the milestone to Backlog?

I am not even sure there's anything to be done here. 

For ICS, off-strategy should be a noop. For TWCS, we reshape in order to segregate data across windows.

Maybe @bhalevy remembers better.

**Comment by bhalevy** [comment]
> > Is this issue still relevant? Can we change the milestone to Backlog?
> 
> I am not even sure there's anything to be done here.
> 
> For ICS, off-strategy should be a noop. For TWCS, we reshape in order to segregate data across windows.
> 
> Maybe [@bhalevy](https://github.com/bhalevy) remembers better.

Right, and if we disable offstrategy, then repair would segregate the data into windows when written.

I'd reconsider this with incremenral repair.
Cc @asias @denesb @tgrabiec

**Comment by bhalevy** [comment]
@denesb who should be assigned here for 2026.1?

**Comment by denesb** [comment]
Re-assigned to @lkshminarayanan
